crates/re_arrow_store/tests/data_store.rs

```diff
@@ -885,6 +889,170 @@ fn gc_impl(store: &mut DataStore) {
     }
 }
 
+#[test]
+fn protected_gc() {
+    init_logs();
+
+    for config in re_arrow_store::test_util::all_configs() {
+        let mut store = DataStore::new(InstanceKey::name(), config.clone());
+        protected_gc_impl(&mut store);
+    }
+}
+
+fn protected_gc_impl(store: &mut DataStore) {
+    init_logs();
+
+    let ent_path = EntityPath::from("this/that");
+
+    let frame0: TimeInt = 0.into();
+    let frame1: TimeInt = 1.into();
+    let frame2: TimeInt = 2.into();
+    let frame3: TimeInt = 3.into();
+    let frame4: TimeInt = 4.into();
```

Comment:
This is a matter of taste, but I usually find this easier to read:

```suggestion
    let frame0 = TimeInt::from(0);
    let frame1 = TimeInt::from(1);
    let frame2 = TimeInt::from(2);
    let frame3 = TimeInt::from(3);
    let frame4 = TimeInt::from(4);
```